File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    #answer = Counter(participant) - Counter(completion)
    #return list(answer.keys())[0]
    
    participant.sort()
    completion.sort()

    for p, c in zip(participant, completion):
        if p != c:
            return p
    print(participant[-1])
    return participant[-1]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> hm = new HashMap<>();
        
        // 동명이인
        for(String person:participant){
            if(hm.containsKey(person)){
                hm.put(person, hm.get(person) + 1);
            } else {
                hm.put(person, 1);
            }
        }
        
        for(String person:completion){
            hm.put(person, hm.get(person) - 1);
        }
        
        for(String person:hm.keySet()){
            if(hm.get(person) != 0){
                return person;
            }
        }
        
        String answer = "";
        return answer;
    }
}